### ISSUE TITLE
`no-extra-parens` Some unnecessary parens in simple assignment statement can't be remove.

### ISSUE THREAD

**Issue by finscn** [eslint-stylistic/eslint-stylistic#185]
### Describe the bug

The orginal code: 
```ts
const a: any = {};
const b: any = 'key';

const c = (b as string);
const d = (b) as (string);
const e = [(b as string)];

const f = a[(b as string)];
```

with the rule  `"@stylistic/no-extra-semi": "error"`, the Expected Result like this:
```ts
const a: any = {};
const b: any = 'key';

const c = b as string;
const d = b as string;
const e = [b as string];

const f = a[b as string];
```

But the Actual Result like this:
```ts
const a: any = {};
const b: any = 'key';

const c = (b as string);
const d = (b) as (string);
const e = [(b as string)];

const f = a[b as string];
```

Unnecessary parens in Line 3/4/5 can NOT be removed.

Only the last line is correct.

### Reproduction

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-8narmc?file=playground%2Feslint.config.js,playground%2Findex.ts

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
